### ISSUE TITLE
Investigate the use of open-source templates and frameworks

### ISSUE THREAD

**Issue by flora8984461** [EclipseFdn/react-eclipsefdn-members#41]
PROS:
- It contains `<Select>` (including Async), `<DatePicker>` Component, which I can take into use and remove "react-select", "react-DatePicker" 
- Its style has some nice effects 

--------------------------------------------------

CONS:
- Need to look into Styles 
- There are some other components that are great, might be useful in the future, but not needed for now.

**Comment by autumnfound** [comment]
Some other things to consider:
- How big is the library for CSS + JS? Its not a huge factor, but if the library is small that could be a bonus for us!
- Does it play well with bootstrap? We have a lot of bootstrap concepts that will likely be on the page we embed the form on, so ideally it would play well with it (this is even considering that we will have a reduced size stylesheet for this page likely)
- How does it factor in accessibility? (Bootstrap form elements have a lot of hooks and examples for accessibility as an example)

**Comment by flora8984461** [comment]
Some notes:
```
Unpacked Size
5.79 MB

Total Files
1560
```
I'll research on `Does it play well with bootstrap?` and `How does it factor in accessibility?` later.

**Comment by flora8984461** [comment]
For `Does it play well with bootstrap?`

I have some initial thoughts:

1) Bootstrap's JS 
Bootstrap's JS is hard to be used in React App. I know we can use jquery within React App, but currently, I need to use `windows.$(selector)` to use it. I am not sure if this is good. 
Normally to embed jquery and bootstrap js in react app, you need `npm install jquery` and `npm install bootstrap` and import them into the components.

Normally using bootstrap with react, would suggest using https://reactstrap.github.io/ or https://react-bootstrap.github.io/getting-started/introduction. But maybe our current existing custom styles will be gone.
 
references: 
https://kingrayhan.medium.com/how-do-i-use-jquery-in-react-app-976c124f5448
https://reactjs.org/docs/integrating-with-other-libraries.html

2) But Bootstrap's html css is okay. Since we already have the css settings align with the theme, I just need to add the classname. Just refer to what I currently have.

3) If using Bootstrap together with Material UI, that's really two separate libraries and needs to be handled differently.

Will take a further look.

**Comment by autumnfound** [comment]
> For `Does it play well with bootstrap?`
> 
> I have some initial thoughts:
> 
> 1. Bootstrap's JS
>    Bootstrap's JS is hard to be used in React App. I know we can use jquery within React App, but currently, I need to use `windows.$(selector)` to use it. I am not sure if this is good.
>    Normally to embed jquery and bootstrap js in react app, you need `npm install jquery` and `npm install bootstrap` and import them into the components.
> 
> Normally using bootstrap with react, would suggest using https://reactstrap.github.io/ or https://react-bootstrap.github.io/getting-started/introduction. But maybe our current existing custom styles will be gone.
> 
> references:
> https://kingrayhan.medium.com/how-do-i-use-jquery-in-react-app-976c124f5448
> https://reactjs.org/docs/integrating-with-other-libraries.html

This isn't a question of using bootstrap or jQuery within react, but on the same page. A lot of bootstrap works on hitting nodes in the document, and doesn't need to be active in the same bundle as react. This can be part of the standard JS bundle we include that already includes that JS.

> 1. But Bootstrap's html css is okay. Since we already have the css settings align with the theme, I just need to add the classname. Just refer to what I currently have.
> 2. If using Bootstrap together with Material UI, that's really two separate libraries and needs to be handled differently.
> 
> Will take a further look.

I'm more concerned about CSS rules fighting and creating ugly UI's, or adding additional rules on top of each other that break the look and feel. Bootstrap has stuff like `row`, `container`, and `btn` in their class definition as an example

**Comment by flora8984461** [comment]
I am going to use this comment to track down references:

https://stackoverflow.com/questions/50831331/bootstrap-vs-material-ui-for-react

**Comment by flora8984461** [comment]
For the current stage, only form involved, we might not quite need it.

For current one, if using Material UI:
The nice things would be `<Select>`, `<TextField>`, `<Button>`, `<Datepicker>` .
The `<Stepper>` is my own and has some functionality for the form. I don't think using Stepper from Material UI is good.

Styling with them is not quite complex.

1) Change my Input component, to use Material UI input + Formik Fields.
2) Change my select component, to use Material UI Select + Formik Fields.
3) Change my Date pick to sue Material UI Date pick + Formik Fields.
4) Add some global styles / reusable styles.

For Later:
I am sure it will be useful for later stages, with the membership portals. Because it has a great dashboard layout and other components, popup, accordion, ... etc., including its own nice material icons.

With Bootstrap:
Actually, Bootstrap has no conflicts. Even try to use the classname from Bootstrap, it won't work on Material UI component. 
They are two separate things.

Material UI itself has `Grid` similar to Boostrap cols. And can change the default breakpoint by customizing it.

Accessibilities 
Accessibilities almost supported, no obvious anti-Accessibilities I found for now.

**Comment by autumnfound** [comment]
**TLDR**: Recommendation is Tabler React, fully OS, based in bootstrap, extensively documented. Provide other options with listed pros/cons.

## Tabler React
Info: https://github.com/tabler/tabler-react
Demo: http://tabler-react.com/

Pros:
- Fully Open source (MIT license)
- Contains decently developed form element styling
- Extensive documentation on components (though light on descriptions/text)

Cons:
- Support is slow, but the author is updating issues in the past week
- Precompiled CSS (we would lose BS variable configs)
- No designer assets available

## Creative Tim themes
Paper Dashboard
Info: https://www.creative-tim.com/product/paper-dashboard-react?partner=104080
Demo: https://demos.creative-tim.com/paper-dashboard-react/?_ga=2.36656672.1672623297.1614191348-1541643693.1614191348#/admin/dashboard

React dashboard
Info: https://www.creative-tim.com/product/light-bootstrap-dashboard-react
Demo: https://demos.creative-tim.com/light-bootstrap-dashboard-react/?&_ga=2.31281471.1672623297.1614191348-1541643693.1614191348#/

Pros: 
- MIT License for the free version
- Bootstrap 4 implementation (responsive)
- Has thorough documentation + a small spread of basic components 
- Provides Sass files for customization (according to docs)

Cons:
- Designer assets available in the pro theme
- Freemium/Pro option doesn't allow for redistribution
> You can’t re-distribute the Item as stock, in a tool or template, or with source files. You can’t do this with an Item either on its own or bundled with other items. You can’t re-distribute or make available the Item as-is or with superficial modifications.

## Pure Material-UI
Info: https://material-ui.com/getting-started/templates/
- Dashboard sample available at https://github.com/mui-org/material-ui/tree/master/docs/src/pages/getting-started/templates/dashboard
Demo: https://material-ui.com/getting-started/templates/dashboard/

Pros: 
- Extensible CSS in code (Examples of this in action is seen on OpenVSX)
- Free to use (MIT)
- Responsive through Material Design framework

Cons:
- Uses a different framework than we have grown accustomed to (Material Design)
- Would require some work to figure out how to inject site variables for alternate colour schemes
- More work as less of the UI is ready to go OOTB

## Devias IO
Info: https://github.com/devias-io/react-material-dashboard
Demo: https://material-kit-react.devias.io/app/dashboard

Pros:
- MIT License for the free version
- Multiple color themes available
- Very similar look and feel to OpenVSX (vanilla Material UI, still very clean)

Cons:
- Poor documentation around components and how they are used
- Sketch/Figma designer files supposedly available for free but links are broken
- Freemium/Pro option doesn't allow for redistribution

## ArchitectUI

Pros:
- MIT License for the free version
- Multiple color themes available
- Contains basic form element theming

Cons:
- Documentation is locked behind the Pro version (unknown quality)
- Components look to be a low-quality implementation of Bootstrap 4
- Does not contain Sass files for the extension of original code in free version
- Freemium/Pro option doesn't allow for redistribution
> You cannot resell, redistribute, lease, license, or offer the product to any third party as it is.

**Comment by flora8984461** [comment]
For the one that using bootstrap 4, my concern would be, since we're using bootstrap 3, and we import it with our theme as well, would there be conflicts of js & css of both bootstrap 4 and 3? 🤔

**Comment by autumnfound** [comment]
> For the one that using bootstrap 4, my concern would be, since we're using bootstrap 3, and we import it with our theme as well, would there be conflicts of js & css of both bootstrap 4 and 3?

It's a good point to make and it would, but these are full-page dashboards, and would fully replace solstice assets in this case.

**Comment by shanda-eclipse** [comment]
From the marketing side, we would like to be able to adjust the font and colors to match the EF style. Other than that, we are pretty flexible on the look of the dashboard.

**Comment by chrisguindon** [comment]
We made a decision: Material UI it is!

**Comment by chrisguindon** [comment]
Thanks @autumnfound for doing the research on this!
